### DISCUSSION TITLE
I can not decide which pipeline is the best

### DISCUSSION THREAD

**Discussion by karabicicim** [satijalab/seurat#8598] [General]
Dear Satijalab,

Recently I found a Nature Com paper, that analyzed CAFs. When ı checked data availability I found that they used this pipeline:

`library(ggthemes)
InstallData("pbmcsca")
data("pbmcsca")
fib_clear <- NormalizeData(fib_clear)
fib_clear <- FindVariableFeatures(fib_clear)
fib_clear <- RunFastMNN(object.list = SplitObject(fib_clear, split.by = "sample"))
fib_clear <- RunUMAP(fib_clear, reduction = "mnn", dims = 1:30) 
fib_clear <- NormalizeData(fib_clear) %>% FindVariableFeatures() %>% ScaleData() %>% RunPCA(verbose = FALSE)
fib_clear.h <- RunHarmony(fib_clear, group.by.vars = "sample",assay.use = "SCT", dims.use = 1:30)
fib_clear.h <- RunUMAP(fib_clear.h, reduction = "harmony", dims = 1:30)
fib_clear.h <- FindNeighbors(fib_clear.h, reduction = "harmony", dims = 1:30) %>% FindClusters()
fib_clear.h <- PercentageFeatureSet(fib_clear.h, pattern = "^KRT", col.name = "percent.krt")
fib_clear.h <- PercentageFeatureSet(fib_clear.h, pattern = "MGP", col.name = "percent.MGP")
fib_clear.h <- SCTransform(fib_clear.h , vars.to.regress = c("mitoRatio","S.Score","G2M.Score", "percent.krt", "percent.MGP"), verbose = TRUE,return.only.var.genes = FALSE)
#PCA
fib_clear.h <- RunPCA(object = fib_clear.h, verbose = FALSE)
ElbowPlot(fib_clear.h)

ElbowPlot(fib_clear.h, ndims =50)

print(x = fib_clear.h[['pca']], dims = 1:5, nfeatures = 5, projected = FALSE)
PCA_genes <- VizDimLoadings(object = fib_clear.h, dims = 1:30)
PCAPlot(object = fib_clear.h)

#Select dimensions
DimHeatmap(object = fib_clear.h, dims = 1, cells = 500, balanced = TRUE)
DimHeatmap(object = fib_clear.h, dims = 15:30, cells = 500, balanced = TRUE)
ElbowPlot(object = fib_clear.h, ndims =50)

#UMAP & TSNE
fib_clear.h <- RunUMAP(object = fib_clear.h, dims = 1:30, reduction = "harmony")
fib_clear.h <- RunTSNE(object = fib_clear.h, dims = 1:30, reduction = "harmony")
`

This pipeline give this results: 
![image](https://github.com/satijalab/seurat/assets/127997586/90268709-c006-4dbe-9b5d-8d47fc474cac)

![image](https://github.com/satijalab/seurat/assets/127997586/9e1f5523-326b-430d-9b0f-86b6c621286d)

And another pipeline generally I follow: After batch correction step I follow:

`
fib_clear_sct <- SCTransform(fib_clear, vst.flavor = "v2", verbose = FALSE) %>%
    RunPCA(npcs = 30, verbose = FALSE) %>%
    RunHarmony(assay.use="SCT", group.by.vars = "sample") %>%
    RunUMAP(reduction = "harmony", dims = 1:30, verbose = FALSE) %>%
    FindNeighbors(reduction = "harmony", dims = 1:30, verbose = FALSE) %>%
    FindClusters(resolution = c(0.065, 0.1, 0.2, 0.3, 0.4, 0.6), verbose = FALSE)
    
`

And the results:
![image](https://github.com/satijalab/seurat/assets/127997586/54b5b89a-ab8e-45ba-a8b5-75707d3e1940)
![image](https://github.com/satijalab/seurat/assets/127997586/6b74e29e-4df3-423c-b378-34ab216873c3)

To be honest second pipeline gives clearer cluster separation but I want to integrate it with other datasets analysed in the paper so I am not sure if should i use the same pipeline or replace it with a new pipeline.

I am kindly asking which one looks better. I will be grateful for any suggestions.

**Comment by igrabski** [comment]
Hi, thanks for your question. We use these issues for addressing technical problems or bugs; for more subjective analysis-based questions, we encourage using the Discussions tab instead -- I'll convert this issue to a discussion.
